localstack/services/opensearch/resource_providers/aws_opensearchservice_domain.py

```diff
@@ -263,28 +275,9 @@ def delete(
           - es:DeleteDomain
           - es:DescribeDomain
         """
-        name = request.desired_state["DomainName"]
-        # LOG.warning(f"deleting domain {request.custom_context=}")
-        assert name is not None
-        if not request.custom_context.get("started", False):
-            # first time in the loop
-            request.aws_client_factory.opensearch.delete_domain(DomainName=name)
-            return ProgressEvent(
-                status=OperationStatus.IN_PROGRESS,
-                resource_model=request.desired_state,
-                custom_context={"started": True},
-            )
-
-        # we have entered the loop again so check the resource status
-        try:
-            request.aws_client_factory.opensearch.describe_domain(DomainName=name)
-            return ProgressEvent(
-                status=OperationStatus.SUCCESS, resource_model=request.desired_state
-            )
-        except request.aws_client_factory.opensearch.exceptions.ResourceNotFoundException:
-            return ProgressEvent(
-                status=OperationStatus.SUCCESS, resource_model=request.desired_state
-            )
+        opensearch_client = request.aws_client_factory.opensearch
+        opensearch_client.delete_domain(DomainName=request.previous_state["DomainName"])
+        return ProgressEvent(status=OperationStatus.SUCCESS, resource_model={})
```

Comment:
Is the delete operation on opensearch synchronous? If not we need the same pattern as in create here.

Reply:
yes, seems to be synchronous.